### PR TITLE
Add the packaging metadata to build the cpp-ethereum snap

### PR DESCRIPTION
**Pull request by elopio**
This is a package for the secure installation of apps that works in most Linux distributions.
Landing it upstream will enable builds that then can be distributed to many users through the Ubuntu store.

### CODE REVIEW

**Comment by elopio** [comment]
You can find more information about snapcraft here: http://snapcraft.io/

This is just an initial snap to start the discussion. It builds, and runs, but needs more testing and polishing.

To test it in an Ubuntu 16.04 machine:

    $ sudo apt install git snapcraft
    $ git clone --recursive https://github.com/elopio/cpp-ethereum
    $ cd cpp-ethereum
    $ git checkout snapcraft
    $ snapcraft
    $ sudo snap install *.snap --dangerous --devmode

(dangerous because the snap you build yourself doesn't come signed from the trusted store. devmode to only warn about confinement, which helps to define the required permissions properly. Once you push it to the store, your users will just have to do $ sudo snap install cpp-ethereum)

If you have any questions or comments, please let me know.

**Comment by chfast** [comment]
Wow, nice. Looks good to me.

Just one question: is the path `snap/snapcraft.yml` the "standard" location for the file?

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+version: master
+summary: blockchain app platform
+description: Ethereum C++ project
+
+grade: devel
+confinement: devmode
+
+apps:
+  bench:
+    command: bench
```
Comment (chfast) [inline snap/snapcraft.yaml]: I'm not sure this should be exported. It is just some internal benchmark tool.

**Comment by elopio** [comment]
Thanks for the review @chfast !

`snap/snapcraft.yml` is the default location. You can actually make snapcraft work from any location as long as the `source` is correct.

**Comment by chriseth** [comment]
That's nice! I would prefer the file to be in e.g. `dist/snap/snapcraft.yaml`, then we can do similar things for debian, etc.

**Comment by elopio** [comment]
@chriseth I've moved the yaml. Please note that this requires snapcraft to be called from dist/snap.

**Pull request merged**